test/common/quic/quic_stat_names_test.cc

```diff
@@ -0,0 +1,29 @@
+#include <string>
+
+#include "common/quic/quic_stat_names.h"
+
+#include "test/mocks/stats/mocks.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Quic {
+
+class QuicStatNamesTest : public testing::Test {
+public:
+  QuicStatNamesTest() : scope_(*symbol_table_), quic_stat_names_(*symbol_table_) {}
+
+  Stats::TestUtil::TestSymbolTable symbol_table_;
+  Stats::TestUtil::TestStore scope_;
+  QuicStatNames quic_stat_names_;
+};
+
+TEST_F(QuicStatNamesTest, QuicConnectionCloseStats) {
+  quic_stat_names_.chargeQuicConnectionCloseStats(scope_, quic::QUIC_NO_ERROR,
+                                                  quic::ConnectionCloseSource::FROM_SELF, false);
+  EXPECT_EQ(
+      1U, scope_.counter("downstream.self.quic_connection_close_error_code_QUIC_NO_ERROR").value());
```

Comment:
can we add checks to one of the quic http integration tests in test/integration/quic_http_integration_test.cc